File: starters/node-api/src/health.ts
import { Request, Response } from "express";
import express from "express";

export interface HealthCheck {
  database?: () => Promise<boolean>;
  redis?: () => Promise<boolean>;
}

export function createHealthRouter(checks: HealthCheck = {}) {
  const router = express.Router();

  // GET /health/live — liveness
  router.get("/live", (_req: Request, res: Response) => {
    res.json({
      status: "ok",
      timestamp: new Date().toISOString(),
      checks: {},
    });
  });

  // GET /health/ready — readiness
  router.get("/ready", async (_req: Request, res: Response) => {
    const results: Record<string, string> = {};
    let allOk = true;

    if (checks.database) {
      try {
        results.database = (await checks.database()) ? "ok" : "error";
        if (results.database === "error") allOk = false;
      } catch {
        results.database = "error";
        allOk = false;
      }
    }

    if (checks.redis) {
      try {
        results.redis = (await checks.redis()) ? "ok" : "error";
        if (results.redis === "error") allOk = false;
      } catch {
        results.redis = "error";
        allOk = false;
      }
    }

    const status = allOk ? "ok" : "degraded";
    const code = allOk ? 200 : 503;
    res.status(code).json({
      status,
      timestamp: new Date().toISOString(),
      checks: results,
    });
  });

  return router;
}

File: starters/python-api/app/health.py
"""Platform health contract: GET /health/live, GET /health/ready."""

from datetime import datetime, timezone
from typing import Callable, Optional

from fastapi import APIRouter

router = APIRouter(prefix="/health", tags=["health"])

CheckFn = Callable[[], bool]


def create_health_checks(
    database: Optional[CheckFn] = None,
    redis: Optional[CheckFn] = None,
) -> dict:
    """Run optional dependency checks."""
    checks = {}
    if database:
        try:
            checks["database"] = "ok" if database() else "error"
        except Exception:
            checks["database"] = "error"
    if redis:
        try:
            checks["redis"] = "ok" if redis() else "error"
        except Exception:
            checks["redis"] = "error"
    return checks


@router.get("/live")
def live():
    """Liveness — process is running."""
    return {
        "status": "ok",
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "checks": {},
    }


@router.get("/ready")
def ready():
    """Readiness — ready for traffic. Add DB/Redis checks when configured."""
    checks = {}
    all_ok = True
    return {
        "status": "ok" if all_ok else "degraded",
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "checks": checks,
    }
